packages/react-components/src/components/ChatMessage/ChatMessageComponentWrapper.tsx

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { MessageStatus } from '@internal/acs-ui-common';
+import React from 'react';
+import { MessageProps, MessageRenderer, MessageThreadStyles, _ChatMessageProps } from '../MessageThread';
+import { ChatMessage, OnRenderAvatarCallback } from '../../types';
+/* @conditional-compile-remove(data-loss-prevention) */
+import { BlockedMessage } from '../../types';
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { FileDownloadHandler, FileMetadata } from '../FileDownloadCards';
+/* @conditional-compile-remove(mention) */
+import { MentionOptions } from '../MentionPopover';
+import { MessageStatusIndicatorProps } from '../MessageStatusIndicator';
+import { FluentChatMessageComponentWrapper } from './FluentChatMessageComponentWrapper';
+import { DefaultSystemMessage } from './DefaultSystemMessage';
+
+/**
+ * @private
+ */
+export type ChatMessageComponentWrapperProps = _ChatMessageProps & {
+  /**
+   * UserId of the current user.
+   */
+  userId: string;
+  styles: MessageThreadStyles | undefined;
+  shouldOverlapAvatarAndMessage: boolean;
+  onRenderMessageStatus: ((messageStatusIndicatorProps: MessageStatusIndicatorProps) => JSX.Element | null) | undefined;
+  defaultStatusRenderer: (
+    message: ChatMessage | /* @conditional-compile-remove(data-loss-prevention) */ BlockedMessage,
+    participantCount: number,
+    readCount: number,
+    status?: MessageStatus
+  ) => JSX.Element;
+  onRenderMessage?: (messageProps: MessageProps, messageRenderer?: MessageRenderer) => JSX.Element;
+  onRenderAvatar?: OnRenderAvatarCallback;
+  showMessageStatus?: boolean;
+  participantCount?: number;
+  readCount?: number;
+  /* @conditional-compile-remove(file-sharing) */
+  onRenderFileDownloads?: (userId: string, message: ChatMessage) => JSX.Element;
+  onActionButtonClick: (
+    message: ChatMessage,
+    setMessageReadBy: (
+      readBy: {
+        id: string;
+        displayName: string;
+      }[]
+    ) => void
+  ) => void;
+  /* @conditional-compile-remove(file-sharing) */
+  fileDownloadHandler?: FileDownloadHandler;
+  /* @conditional-compile-remove(date-time-customization) */
+  onDisplayDateTimeString?: (messageDate: Date) => string;
+  /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+  onFetchInlineAttachment: (attachments: FileMetadata[], messageId: string) => Promise<void>;
+  /* @conditional-compile-remove(image-gallery) */
+  onInlineImageClicked?: (attachmentId: string, messageId: string) => Promise<void>;
+  /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+  inlineAttachments: Record<string, Record<string, string>>;
+  /* @conditional-compile-remove(mention) */
+  mentionOptions?: MentionOptions;
+};
+
+/**
+ * @private
+ */
+export const ChatMessageComponentWrapper = (props: ChatMessageComponentWrapperProps): JSX.Element => {
```

Comment:
Would be good to have some comments around why this component exists.

Reply:
Added!